godot-core/src/private.rs

```diff
@@ -261,19 +275,76 @@ pub(crate) fn has_error_print_level(level: u8) -> bool {
     ERROR_PRINT_LEVEL.load(atomic::Ordering::Relaxed) >= level
 }
 
+#[cfg(debug_assertions)]
+struct ScopedFunctionStack(Vec<*const dyn Fn() -> String>);
+
+#[cfg(debug_assertions)]
+impl ScopedFunctionStack {
+    /// # Safety
+    /// Function must removed (using pop_function) before lifetime is invalidated.
+    unsafe fn push_function(&mut self, function: &dyn Fn() -> String) {
+        /// # Safety
+        /// The caller must ensure that the function isn't used past its original lifetime.
+        /// (Invariant must be held by push_function)
+        unsafe fn assume_static_lifetime(
+            value: &dyn Fn() -> String,
+        ) -> &'static dyn Fn() -> String {
+            std::mem::transmute(value)
+        }
+        self.0.push(assume_static_lifetime(function) as *const _);
+    }
+
+    fn pop_function(&mut self) {
+        self.0.pop();
+    }
```

Comment:
Silently returns `None` if called on empty stack -- should this use `expect()`?

(which might then cause a panic inside the handler, but this is a severe bug that shouldn't happen, so it's probably OK)